processor/batchprocessor/batch_processor_test.go

```diff
@@ -611,6 +611,20 @@ func BenchmarkBatchMetricProcessor(b *testing.B) {
 		Timeout:       100 * time.Millisecond,
 		SendBatchSize: 2000,
 	}
+	runMetricsProcessorBenchmark(b, cfg)
+}
+
+func BenchmarkMultiBatchMetricProcessor(b *testing.B) {
+	b.StopTimer()
+	cfg := Config{
+		Timeout:       100 * time.Millisecond,
+		SendBatchSize: 2000,
+		MetadataKeys:  []string{"test", "test2"},
+	}
+	runMetricsProcessorBenchmark(b, cfg)
```

Comment:
We should change the `runMetricsProcessorBenchmark` to use `b.RunParralel(func(pb *testing.PB) {...})`, because for example in case of https://github.com/open-telemetry/opentelemetry-collector/pull/7714 that was the main issue.

Reply:
Switched the method to use RunParallel. PTAL